scripts/variantstore/wdl/GvsCallsetStatistics.wdl

```diff
@@ -0,0 +1,707 @@
+        String dataset_name
+        String filter_set_name
+        String extract_prefix
+        String metrics_table
+        Int chromosome
+    }
+    command <<<
+        set -o errexit -o nounset -o xtrace -o pipefail
+
+        echo "project_id = ~{project_id}" > ~/.bigqueryrc
+
+        bq query --location=US --project_id=~{project_id} --format=csv --use_legacy_sql=false '
+            SELECT COUNT(*) from `~{project_id}.~{dataset_name}.~{metrics_table}` WHERE chromosome = ~{chromosome}
+        ' | sed 1d > existing_row_count.txt
+
+        existing_row_count=$(cat existing_row_count.txt)
+
+        if [ $existing_row_count -gt 0 ]; then
+            echo "Found $existing_row_count rows in '~{project_id}.~{dataset_name}.~{metrics_table}' for chromosome '~{chromosome}', exiting."
+            exit 1
+        fi
+
+        bq query --location=US --project_id=~{project_id} --use_legacy_sql=false '
+        CREATE TEMPORARY FUNCTION titv(ref STRING, allele STRING)
+        RETURNS STRING
+            LANGUAGE js AS """
+                if ( ref.length > 1 || allele.length > 1) {
+                    return "other";
+                } else if ( (ref == "A" && allele == "G") ||
+                            (ref == "G" && allele == "A") ||
+                            (ref == "C" && allele == "T") ||
+                            (ref == "T" && allele == "C") ) {
+                    return "ti";
+                } else {
+                    return "tv";
+                }
+        """;
+
+        CREATE TEMPORARY FUNCTION type(ref STRING, allele STRING, gt_str STRING)
+        RETURNS STRING
+            LANGUAGE js AS """
+
+        alts = allele.split(",")
+
+        // get the the non-reference allele indexes
+        ai = gt_str.replace("|","/").split("/").filter(i => i != "0");
+
+        // the the distinct set of lengths of the alternates
+        alt_lengths = new Set(ai.map(i => alts[parseInt(i)-1].length))
+
+        if (alt_lengths.size > 1) {
+            return "complex"
+        } else {
+            // get first (only) element
+            al = alt_lengths.keys().next().value
+
+            if ( ref.length == al && al == 1) {
+                return "snp"
+            } else if (ref.length > al) {
+                return "del"
+            } else if (ref.length < al) {
+                return "ins"
+            } else {
+                return "other"
+            }
+        }
+        """;
+
+        INSERT `~{project_id}.~{dataset_name}.~{metrics_table}` (
+            filter_set_name,
+            sample_id,
+            chromosome,
+            variant_entries,
+            del_count,
+            ins_count,
+            snp_count,
+            ti_count,
+            tv_count,
+            snp_het_count,
+            snp_homvar_count,
+            indel_het_count,
+            indel_homvar_count,
+            singleton,
+            pass_qc
+        )
+        SELECT "~{filter_set_name}" filter_set_name,
```

Comment:
I know that none of the explanations are in the code that you are looking at to write this wdl, but I think getting Lee to add some context about what is being calculated would be really helpful. I'm fine with that being a future ticket